### PR TITLE
Convert product-elements/button to TypeScript

### LINKED ISSUE #7090: Convert `product-elements/button` to TypeScript
Folder: https://github.com/woocommerce/woocommerce-blocks/tree/trunk/assets/js/atomic/blocks/product-elements/button

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #7090

Currently, the `product-elements/button` is still written in JS and is using [prop-types](https://www.npmjs.com/package/prop-types) for type checking. This PR aims to convert `product-elements/button` to TypeScript.

#### User Facing Testing

1. Create a test page and add the `Cart block` to it.
2. Verify that the `Cart Cross-Sells inner block` exists and that it contains buttons.
3. Make sure to have one test product that contains a cross-sells product.
4. Add that product to the cart and open the cart.
5. Verify that the cross-sells product can be added to the cart.

* [x] Do not include in the Testing Notes

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/button/attributes.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-export const blockAttributes = {
+/**
+ * External dependencies
+ */
+import { BlockAttributes } from '@wordpress/blocks';
```
Comment (alexflorisca) [inline assets/js/atomic/blocks/product-elements/button/attributes.ts]:
```suggestion
import type { BlockAttributes } from '@wordpress/blocks';
```

File: assets/js/atomic/blocks/product-elements/button/types.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export interface BlockAttributes {
```
Comment (alexflorisca) [inline assets/js/atomic/blocks/product-elements/button/types.ts]: nitpick: move this down so all exported interfaces are together, at the bottom of the file. It reads more naturally that way

**Review by alexflorisca — COMMENTED** [review]
Nice work Niels! Left a few comments and I noticed there are a few type errors in block.tsx.

The one on line 67, the `useStoreAddToCart` only takes one parameter `id` but we pass it two arguments. I think we can safely remove the second argument.

The one on line 110 we just need to cast the result of the `getSetting` using `as` whatever the shape of `cartRedirectAfterAdd` is. 

The last one I think we can leave, it's complaining about overloaded className prop which is legit.

File: assets/js/atomic/blocks/product-elements/button/types.ts (hunk 2 above)
Comment (nielslange) [reply assets/js/atomic/blocks/product-elements/button/types.ts]: That's a great suggestion. Thanks for that, @alexflorisca!

**Review by alexflorisca — APPROVED** [review]
Nice looks good Niels, thanks for addressing the feedback. One last nitpicky bit is that I don't think there's a need to define the `CartRedirectAfterAdd` interface separately as it's just an object with a key inside it. Might be a little clearer here if you define this inline. Not a blocker though, nice work.

**Comment by nielslange** [comment]
Thanks for your review and your nitpicks, @alexflorisca. Especially the nitpicks are more than welcome as this helps me to grow as a developer. I've addressed your last suggestion and merged the PR accordingly. 🙌

**Pull request merged**